### ISSUE TITLE
Hardcoded kubernetes-version?

### ISSUE THREAD

**Issue by ahmetb** [kelseyhightower/kubeadm-single-node-cluster#3]
Looks like https://github.com/kelseyhightower/kubeadm-single-node-cluster/blob/21b00565867fc5b36375fa374bb863fe6e5b648a/startup.sh#L42

is hardcoding the value. It looks like the provided gcloud commands + startup.sh will bring the value to KUBERNETES_VERSION env var anyway:

https://github.com/kelseyhightower/kubeadm-single-node-cluster/blob/21b00565867fc5b36375fa374bb863fe6e5b648a/startup.sh#L47-L50

I'm not sure if KUBERNETES_VERSION env will override the hardcoded value in MasterConfiguration. The [kubeadm docs](https://kubernetes.io/docs/admin/kubeadm/) say it doesn't:

> Note: When providing configuration values using both a configuration file and flags, the file will take precedence. For example, if a file exists with:

In this case, could the hardcoded value actually be causing kubeadm to always provision `stable-1.7` regardless of the version saved in gce metadata key?

**Comment by ahmetb** [comment]
I'm trying to get a `v1.8.0-beta.1` cluster with this.

So far, I tried the following, but I'm still getting v1.7.5 cluster.

1. I forked this repo and replaced the hardcoded version value in kubeadm.conf to  `kubernetesVersion: v1.8.0-beta.1`, as well as created instance with `--metadata kubernetes-version=v1.8.0-beta.1,...`. No luck.
     - (I can see the modified startup script correctly being picked up in `cat /kubeadm.conf`)

2. I removed `kubernetesVersion` in kubeadm.conf and used `--metadata kubernetes-version=v1.8.0-beta.1,...` since kubeadm is supposed to pick up the KUBERNETES_VERSION env when it's not listed in the MasterConfiguration manifest. No luck here, either.

/cc @luxas could it be possibly something wrong with the format of the version value that gets silently ignored and defaulted to latest stable (v1.7.5) somehow?
